### DISCUSSION TITLE
Why `--emitDeclarationOnly` in TypeScript plugin?

### DISCUSSION THREAD

**Discussion by alecmev** [paularmstrong/onerepo#684] [Q&A] [answered]
I like `useProjectReferences` a lot, as it also lists transitive dependencies (which [workspaces-to-typescript-project-references](https://github.com/azu/monorepo-utils/tree/master/packages/%40monorepo-utils/workspaces-to-typescript-project-references) doesn't do, for example). However, it comes bundled with `--emitDeclarationOnly`:

https://github.com/paularmstrong/onerepo/blob/be882795966edd469524734de8140be6fc111685/plugins/typescript/src/commands/typescript.ts#L80

Why does it deviate from the non-project-references branch, which uses `--noEmit`? This breaks for me because my root-level `tsconfig.json` has `noEmit` in it, and these two options are incompatible with each other.

In general, why are these options forced at all? At first glance, it doesn't seem dangerous to emit, if desired, and if not, it's as easy as putting `noEmit` in a base config.

`--build` can probably be an independent option too, no? It does depend on project references, but project references don't depend on it. In my case, I'm thinking about not using the build mode at all, because now oneRepo orchestrates things. And if I'm not using `--build` then maybe the root `tsconfig.json` could also be skipped when updating project references.

**Comment by paularmstrong** [comment, marked as answer]
> Why does it deviate from the non-project-references branch, which uses --noEmit? This breaks for me because my root-level tsconfig.json has noEmit in it, and these two options are incompatible with each other.

project references require declarations to benefit from using the references/ts cache, but the declaration does nothing in the other branch.

> In general, why are these options forced at all? At first glance, it doesn't seem dangerous to emit, if desired, and if not, it's as easy as putting noEmit in a base config.

Emit is unnecessary when using [source dependencies](https://deploy-preview-680--onerepo.netlify.app/docs/source-dependencies/)

> --build can probably be an independent option too, no?

Related discussion: https://github.com/paularmstrong/onerepo/discussions/666
